### DISCUSSION TITLE
Sound reactive WS2812B LED strip with bluetooth input

### DISCUSSION THREAD

**Discussion by JoaoPedroAssis** [pschatzmann/ESP32-A2DP#303] [Show and tell]
Hey there!! I made a sound reactive system based on the ESP32, that performs the FFT analysis of a music sample obtained via Bluetooth.

Special thanks to @pschatzmann for making this amazing library and for answering all my dumb questions 😆 , this project probably wouldn't be possible without it

This demo was made using the FreeRTOS to schedule the bluetooth and FFT tasks, and the Espressif RingBuffer implementation to share data between the tasks. The result of the FFT analysis was animated using the `fill_noise16()` function from the FastLED library, along with other methods. The system also outputs the music via I2S, which feeds an external DAC with AUX cable output.

Hope y'all like it! Greetings from Brazil  🇧🇷 !

https://user-images.githubusercontent.com/38732089/193065050-43272ff9-3b9d-45c8-9602-de275c2e0d7d.mp4

**Comment by t413** [comment]
Super impressive @JoaoPedroAssis! Love it! And looks like this is the code repository from your profile [JoaoPedroAssis/smartleds](https://github.com/JoaoPedroAssis/smartleds)
